src/plugins/timelist/TimelistComponent.vue

```diff
@@ -290,48 +377,53 @@ export default {
             callback: () => {
               const oldTelemetryObject = this.planObjects[0];
               this.removeFromComposition(oldTelemetryObject);
-              this.addItem(telemetryObject);
+              this.addItem(planObject);
               dialog.dismiss();
             }
           },
           {
             label: 'Cancel',
             callback: () => {
-              this.removeFromComposition(telemetryObject);
+              this.removeFromComposition(planObject);
               dialog.dismiss();
             }
           }
         ]
       });
     },
-    removeFromComposition(telemetryObject) {
-      this.composition.remove(telemetryObject);
+    removeFromComposition(planObject) {
+      this.composition.remove(planObject);
     },
     removeItem() {
       this.planObjects = [];
       this.resetPlanData();
     },
     resetPlanData() {
       this.planData = {};
+      this.groups = [];
+      this.planActivities = [];
+      this.sortedItems = [];
     },
     getPlanData(domainObject) {
+      this.resetPlanData();
       this.planData = getValidatedData(domainObject);
-    },
-    listActivities() {
-      let groups = getValidatedGroups(this.domainObject, this.planData);
-      let activities = [];
-
-      groups.forEach((key) => {
+      this.groups = getValidatedGroups(this.domainObject, this.planData);
+      this.groups.forEach((key) => {
         if (this.planData[key] === undefined) {
           return;
         }
         // Create new objects so Vue 3 can detect any changes
-        activities = activities.concat(JSON.parse(JSON.stringify(this.planData[key])));
+        this.planActivities = this.planActivities.concat(
+          JSON.parse(JSON.stringify(this.planData[key]))
+        );
```

Comment:
I think we might just be able to do this:

```js
this.planActivities.push(...this.planData[key]);
```

Unless we need a deep clone? I don't think we do.

Reply:
Hmm. I think we do? I'd like to leave it for now if that's ok? I'll come back and test it later.

Reply:
Nah, let's change this. JSON parse / stringify is costly anyway.